models.ts

```diff
@@ -8,11 +11,53 @@ class Peli {
 }
 
 class PelisCollection {
-  getAll(): Promise<Peli[]> {
-    return jsonfile("...laRutaDelArchivo").then(() => {
-      // la respuesta de la promesa
-      return [];
+  pelis:Peli[] = [];
+  getAll(): Promise<Peli[]>{//listo
+    return jsonfile.readFile("./pelis.json").then((lista) => {
+      this.pelis = lista;
+      return lista
+    });
+  }
+  getById(id:number){//listo
+    return this.getAll().then((lista)=>{
+      const encontrado = find(lista,(i)=>{return i.id == id;})
+      return encontrado;
+    })
+  }
+  search(options:any):Promise<any>{//listo
+    if(options.tag && options.title){
+      return this.getAll().then((lista)=>{
+        const listaAMostrar = filter(lista,(i)=>{
+          return i.title.includes(options.title) && i.tags.includes(options.tag)
+        })
+        return listaAMostrar;
+      });
+    }else if(options.title){
+      return this.getAll().then((lista)=>{
+        const resultado = filter(lista,(i)=>{return i.title.includes(options.title)})
+        return resultado;
+        });
+    }else if(options.tag){
+      return this.getAll().then((lista)=>{
+        const resultado = filter(lista,(i)=>{return i.tags.includes(options.tag)})
+        return resultado;
+      });
+    }    
+  }
```

Comment:
Me parece que es mas simple si solo haces una vez `return this.getAll().then` y dentro de los corchetes al resolver la promesa, creas los tres ifs que ven las opciones que te llegan, igualmente es un detalle y esta bien asi también!